src/Microsoft.Identity.Client/ApiConfig/Parameters/AcquireTokenCommonParameters.cs

```diff
@@ -25,18 +25,16 @@
 // 
 // ------------------------------------------------------------------------------
 
-namespace Microsoft.Identity.Client.ApiConfig
+using System.Collections.Generic;
+using Microsoft.Identity.Client.TelemetryCore;
+
+namespace Microsoft.Identity.Client.ApiConfig.Parameters
 {
-    /// <summary>
-    /// </summary>
-    internal interface IAcquireTokenOnBehalfOfParameters : IAcquireTokenCommonParameters
+    internal class AcquireTokenCommonParameters
     {
-        /// <summary>
-        /// </summary>
-        UserAssertion UserAssertion { get; }
-
-        /// <summary>
-        /// </summary>
-        bool SendX5C { get; }
+        public ApiEvent.ApiIds ApiId { get; set; } = ApiEvent.ApiIds.None;
```

Comment:
>None [](start = 69, length = 4)

What is the story for having the ID's set for each scenario? Does this get set at the AquireToken call or how/when?

Reply:
Each of the classes deriving from RequestBase used to take ApiId as a constructor parameter to pass into RequestBase.  And we did the logic for determining the API ID within the constructor of the request.  But that was all introspection into the specific parameters to determine the API ID.  So now the builder determines the API ID so we can more easily unit test that "these parameters from the user will result in this api id and behavior"

---
In reply to: [249085178](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/798#discussion_r249085178) [](ancestors = 249085178)